src/subscribers/abstract_subscriber.jl

```diff
@@ -65,17 +65,35 @@ struct SubscribedMessage
     sub::Subscriber          # Note this is an abstract type
     lock::ReentrantLock
     name::String
+    task::Vector{Task}
 
     function SubscribedMessage(
         msg::ProtoBuf.ProtoType,
         sub::Subscriber;
         name = getname(sub),
     )
-        new(msg, sub, ReentrantLock(), name)
+        new(msg, sub, ReentrantLock(), name, Task[])
     end
 end
-subscribe(submsg::SubscribedMessage) = subscribe(submsg.sub, submsg.msg, submsg.lock)
-getname(submsg::SubscribedMessage) = submsg.name
+@inline subscribe(submsg::SubscribedMessage) = subscribe(submsg.sub, submsg.msg, submsg.lock)
+@inline getname(submsg::SubscribedMessage) = submsg.name
+@inline getcomtype(submsg::SubscribedMessage) = getcomtype(submsg.sub)
+isrunning(submsg::SubscribedMessage) = !isempty(submsg.task) && !istaskdone(submsg.task[end])
+
+function launchtask(submsg::SubscribedMessage)
+    push!(submsg.task, @async subscribe(submsg))
+    return submsg.task[end]
+end
```

Comment:
This enables you to launch multiple subscriber threads in the same process, when is this necessary? Specifically why does it have to push to a vector of tasks instead of simply a single task?

Reply:
2 reasons: 
1. I wanted to keep the `SubscribedMessage` struct immutable (so the user can't randomly switch out a message or subscriber)
2. It makes it slightly more robust since it'll always append newly launched tasks to the end, instead of overwriting the previous task, which reduces the risk of losing the handle to a running task. I haven't implemented the logic to clear out the previous tasks yet, but presumably that should be done when the node closes.